pkg/kubelet/cm/dra/manager.go

```diff
@@ -240,19 +240,45 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 			continue
 		}
 
+		// Query claim object from the API server
+		resourceClaim, err := m.kubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Get(
```

Comment:
Just thinking out loud....

So now the code relies on:
- The ResourceClaim still existing - guaranteed because the claim is still in use.
- Connectivity to the apiserver - will be needed also for updating the pod status, so okay.
- The allocation not being modified - not meant to be done, but not enforced either. Perhaps add that to https://github.com/kubernetes/kubernetes/pull/116332?

Reply:
All of this is true, yes.

Reply:
Validation added to https://github.com/kubernetes/kubernetes/pull/116332 as suggested.